File: gradle-plugin/karakum/src/jsTest/resources/topLevel/karakum/plugins/blankOutExportStatement.js
import ts from "typescript"

export default (node) => {
    const sourceFileName = node.getSourceFile()?.fileName ?? "generated.d.ts"

    if (
        sourceFileName.endsWith("/myFunction1.d.ts")
        && ts.isExportAssignment(node)
    ) {
        return ""
    }

    return null
}

File: karakum/src/jsTest/functional/io/github/sgrishchenko/karakum/topLevel/plugins/blankOutExportStatement.kt
package io.github.sgrishchenko.karakum.topLevel.plugins

import io.github.sgrishchenko.karakum.extension.createPlugin
import io.github.sgrishchenko.karakum.util.getSourceFileOrNull
import typescript.isExportAssignment

val blankOutExportStatement = createPlugin plugin@{ node, _, _ ->
    val sourceFileName = node.getSourceFileOrNull()?.fileName ?: "generated.d.ts"

    if (
        sourceFileName.endsWith("/myFunction1.d.ts")
        && isExportAssignment(node)
    ) {
        return@plugin ""
    }

    null
}
